CRM/Dedupe/Merger.php

```diff
@@ -619,19 +619,19 @@ public static function retrieveFields($main, $other) {
    *
    * @return array|bool
    */
-  public static function batchMerge($rgid, $gid = NULL, $mode = 'safe', $batchLimit = 1, $isSelected = 2, $criteria = array(), $checkPermissions = TRUE) {
-    $redirectForPerformance = ($batchLimit > 1) ? TRUE : FALSE;
-    $reloadCacheIfEmpty = (!$redirectForPerformance && $isSelected == 2);
+  public static function batchMerge($rgid, $gid = NULL, $mode = 'safe', $batchLimit = 0, $isSelected = 2, $criteria = array(), $checkPermissions = TRUE) {
+    // Not totally clear on the logic here but it suggests the cache is reloaded if we are not dealing with specificially selected contacts.
+    $reloadCacheIfEmpty = ($batchLimit <= 1 && $isSelected == 2);
     $dupePairs = self::getDuplicatePairs($rgid, $gid, $reloadCacheIfEmpty, $batchLimit, $isSelected, '', ($mode == 'aggressive'), $criteria, $checkPermissions);
 
     $cacheParams = array(
       'cache_key_string' => self::getMergeCacheKeyString($rgid, $gid, $criteria, $checkPermissions),
       // @todo stop passing these parameters in & instead calculate them in the merge function based
       // on the 'real' params like $isRespectExclusions $batchLimit and $isSelected.
       'join' => self::getJoinOnDedupeTable(),
-      'where' => self::getWhereString($batchLimit, $isSelected),
+      'where' => self::getWhereString($isSelected),
```

Comment:
This is one of 2 places getWhereString is called from - pass batchLimit to the merge function & on to the retrieval function instead...